src/poliastro/twobody/events.py

```diff
@@ -2,31 +2,78 @@
 from numpy.linalg import norm
 
 
-class LithobrakeEvent:
-    """Terminal event that detects impact with the attractor surface.
+class Event:
+    """Base class for event functionalities.
 
     Parameters
     ----------
-    R : float
-        Radius of the attractor.
+    terminal: bool
+        Whether to terminate integration if this event occurs.
+    direction: float
+        Handle triggering of event.
 
     """
 
-    def __init__(self, R):
-        self._R = R
+    def __init__(self, terminal, direction):
+        self._terminal, self._direction = terminal, direction
         self._last_t = None
 
     @property
     def terminal(self):
-        # Tell SciPy to stop the integration at H = R (impact)
-        return True
+        return self._terminal
+
+    @property
+    def direction(self):
+        return self._direction
 
     @property
     def last_t(self):
         return self._last_t * u.s
 
+    def __call__(self, t, u, k):
+        raise NotImplementedError
+
+
+class AltitudeCrossEvent(Event):
+    """Detect if a satellite crosses a specific threshold altitude.
+
+    Parameters
+    ----------
+    R: ~astropy.units.Quantity
+        Radius of the attractor.
+    thresh_alt: ~astropy.units.Quantity
+        Threshold altitude (in km), defaults to 100 km.
+
+    """
+
+    # Orekit's API stops propagation when descending, but not when ascending.
+    def __init__(self, R, terminal=True, direction=0, thresh_alt=100 * u.km):
+        super().__init__(terminal, direction)
+        self._R = R.to(u.km).value
+        self._thresh_alt = thresh_alt.to(
+            u.km
+        ).value  # Threshold height from the ground.
+
     def __call__(self, t, u, k):
         self._last_t = t
         H = norm(u[:3])
```

Comment:
Let us rename variable `H` to `r_norm` in order no to confuse it with height.

Reply:
Thanks!